src/lib.nr

```diff
@@ -245,7 +245,9 @@ where
     }
 }
 
-trait BigCurveTrait {
+pub trait BigCurveTrait<BigNum> {
```

Comment:
I'm reluctant to introduce a generic here, just to support this.

The `BigCurve` struct already has `pub` members `x` and `y`. Why couldn't you access those, sorry?